__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1508 @@
+            blockMocks.datasetSessionNode, null, undefined, undefined, blockMocks.imperativeProfile);
+
+        mocked(fs.existsSync).mockReturnValueOnce(false);
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Yes" as any);
+        const deleteSpy = jest.spyOn(blockMocks.mvsApi, "deleteDataSet");
+
+        await dsActions.deleteDataset(node, blockMocks.testDatasetTree);
+
+        expect(mocked(fs.unlinkSync)).not.toBeCalled();
+        expect(deleteSpy).toBeCalledWith(node.label);
+    });
+    it("Checking common PS dataset failed deletion attempt due to absence on remote", async () => {
+        globals.defineGlobals("");
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("HLQ.TEST.NODE", vscode.TreeItemCollapsibleState.None,
+            blockMocks.datasetSessionNode, null, undefined, undefined, blockMocks.imperativeProfile);
+
+        mocked(fs.existsSync).mockReturnValueOnce(true);
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Yes" as any);
+        const deleteSpy = jest.spyOn(blockMocks.mvsApi, "deleteDataSet");
+        deleteSpy.mockRejectedValueOnce(Error("not found"));
+
+        await expect(dsActions.deleteDataset(node, blockMocks.testDatasetTree)).rejects.toEqual(Error("not found"));
+
+        expect(mocked(vscode.window.showInformationMessage)).toBeCalledWith("Unable to find file: " + node.label + " was probably already deleted.");
+    });
+    it("Checking common PS dataset failed deletion attempt", async () => {
+        globals.defineGlobals("");
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("HLQ.TEST.NODE", vscode.TreeItemCollapsibleState.None,
+            blockMocks.datasetSessionNode, null, undefined, undefined, blockMocks.imperativeProfile);
+
+        mocked(fs.existsSync).mockReturnValueOnce(true);
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Yes" as any);
+        const deleteSpy = jest.spyOn(blockMocks.mvsApi, "deleteDataSet");
+        deleteSpy.mockRejectedValueOnce(Error(""));
+
+        await expect(dsActions.deleteDataset(node, blockMocks.testDatasetTree)).rejects.toEqual(Error(""));
+        expect(mocked(vscode.window.showErrorMessage)).toBeCalledWith(" Error");
+    });
+    it("Checking PS deletion attempt which was rejected by user in the process", async () => {
+        globals.defineGlobals("");
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("HLQ.TEST.NODE", vscode.TreeItemCollapsibleState.None,
+            blockMocks.datasetSessionNode, null, undefined, undefined, blockMocks.imperativeProfile);
+
+        mocked(fs.existsSync).mockReturnValueOnce(true);
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("No" as any);
+        const deleteSpy = jest.spyOn(blockMocks.mvsApi, "deleteDataSet");
+        deleteSpy.mockClear();
```

Comment:
Do we really need the spy, I would expect we can get rid of them and override the mocks instead.